teuthology/misc.py

```diff
@@ -21,6 +21,8 @@
 import re
 import pprint
 
+from io import BytesIO
```

Comment:
> For python3 compatibility it is preferable to use BytesIO over StringIO.

why? i think as their names suggest, `StringIO` is for reading/writing strings, while `BytesIO` for bytes. why is this difference related to python3?

also, per https://configobj.readthedocs.io/en/latest/configobj.html#configobj-specifications

> A StringIO instance or file object, or any object with a read method. The filename attribute of your ConfigObj will be None 

so i think `StringIO` is better?

probably we could just pass `path` to it?

Reply:
@tchaikov in this particular case there was used cStringIO.StringIO, which does not exists in python3, as well as StringIO.StringIO. The only StringIO in py3 is io.StringIO which is for Unicode, the config file we are using is in ascii/utf-8 encoding, which BytesIO more appropriate. So what is the best?

Reply:
> probably we could just pass `path` to it?

you mean just pass the filename to ConfigObj? this looks to me as a good idea and I thought about it, but there is a `.format` invocation. Though I can not find a `testdir` in 'ceph.conf.template' I thought someone may wish it in the future.

Reply:
@tchaikov Do you want me just do like that:
```
conf = ConfigObj.ConfigObj(path, file_error=True)
```
Instead of handling all that streaming soup?

Reply:
@tchaikov like this? #1323

Reply:
One more else why I used `io.BytesIO` instead of `cStringIO.StringIO`, because
https://docs.python.org/2/library/stringio.html (7.6. cStringIO — Faster version of StringIO):
```
 cStringIO.StringIO([s])
     Return a StringIO-like stream for reading or writing.
     . . .
     Unlike the StringIO module, this module is not able to accept Unicode 
     strings that cannot be encoded as plain ASCII strings.
```
The `io.BytesIO` is present in both py2 and py3.  Any thoughts?

Reply:
@kshtsk yeah, i'd prefer #1323 than this change.